src/scripts/utilities.js

```diff
@@ -35,3 +35,38 @@ export const findObjectsWithNames = (rootRegion, names, regionPath, transverse)
   }
   return targetObjects;
 }
+
+export const getAllGroupNames = (scene) => {
+  let groupNames = [];
+  scene.forEachGeometry((geometry) => {
```

Comment:
More efficient to =>

`scene.getRootRegion().getAllObjects(transverse);`

then iterates through that.